packages/graphql/src/schema/get-where-fields.ts

```diff
@@ -126,17 +126,26 @@ function getWhereFields({
             }
 
             if (["String", "ID"].includes(f.typeMeta.name)) {
-                if (enableRegex) {
+                const matchesSetting: boolean | undefined = features?.filters?.[f.typeMeta.name]?.MATCHES;
+                if (matchesSetting !== undefined) {
+                    if (matchesSetting === true) {
+                        res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
+                    }
+                } else if (enableRegex) {
+                    // Deprecated. To be removed.
                     res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
                 }
 
                 const stringWhereOperators = ["_CONTAINS", "_STARTS_WITH", "_ENDS_WITH"];
 
                 const stringWhereOperatorsNegate = ["_NOT_CONTAINS", "_NOT_STARTS_WITH", "_NOT_ENDS_WITH"];
 
-                Object.entries(features?.filters?.String || {}).forEach(([key, value]) => {
-                    if (value) {
-                        stringWhereOperators.push(`_${key}`);
+                Object.entries(features?.filters?.String || {}).forEach(([filter, enabled]) => {
```

Comment:
Could we not use this loop to add the `MATCHES` filter, especially on the 4.0.0 branch when `enableRegex` is out of the way?

Reply:
I definitely want to do this but I would like to do it in the `4.0.0` branch as you say.